libcanard/canard.c

```diff
@@ -156,7 +156,7 @@ typedef struct TxItem
     //  - Make the payload pointer point to the remainder of the allocated memory following this structure.
     //    The pointer is bad because it requires us to use pointer arithmetics.
     //  - Use a separate memory allocation for data. This is terribly wasteful (both time & memory).
-    uint8_t payload_buffer[];  // NOSONAR
+    uint8_t payload_buffer[CANARD_MTU_CAN_FD];  // NOSONAR
```

Comment:
1. The comment above is no longer relevant and should be either edited or removed.
2. Can we please introduce a new definition in the header file like this in order to remove the implicit dependency on the set of supported transports here:

```c
#define CANARD_MTU_MAX CANARD_MTU_CAN_FD
```